src/dtx/dtx_common.c

```diff
@@ -2016,6 +2016,8 @@ dtx_leader_exec_ops(struct dtx_leader_handle *dlh, dtx_sub_func_t func,
 exec:
 	/* Then execute the local operation */
 	rc = func(dlh, func_arg, -1, NULL);
+	if (dlh->dlh_result == 0)
+		dlh->dlh_result = rc;
```

Comment:
The issue is that as long as "rc" is "-DER_NONEXIST" here, then dtx_leader_exec_ops() will return "-DER_NONEXIST" via L2030. That will overwrite obj_punch_agg_cb().

Reply:
yes, you are right. let me update the patch.